File: PS Multi Tools/Classes/PKGDownloadListViewItem.vb
Imports System.ComponentModel
Imports System.Net
Imports System.Runtime.CompilerServices

Public Class PKGDownloadListViewItem

    Implements INotifyPropertyChanged

    Private _PackageTitleID As String
    Private _PackageName As String
    Private _PackageSize As String
    Private _PackageDownloadState As String
    Private _AssociatedWebClient As WebClient
    Private _PackageContentID As String
    Private _PackageDownloadDestination As String
    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

    Protected Sub NotifyPropertyChanged(<CallerMemberName> Optional propertyName As String = "")
        RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
    End Sub

    Public Property AssociatedWebClient As WebClient
        Get
            Return _AssociatedWebClient
        End Get
        Set
            _AssociatedWebClient = Value
            NotifyPropertyChanged("AssociatedWebClient")
        End Set
    End Property

    Public Property PackageContentID As String
        Get
            Return _PackageContentID
        End Get
        Set
            _PackageContentID = Value
            NotifyPropertyChanged("PackageContentID")
        End Set
    End Property

    Public Property PackageTitleID As String
        Get
            Return _PackageTitleID
        End Get
        Set
            _PackageTitleID = Value
            NotifyPropertyChanged("PackageTitleID")
        End Set
    End Property

    Public Property PackageName As String
        Get
            Return _PackageName
        End Get
        Set
            _PackageName = Value
            NotifyPropertyChanged("PackageName")
        End Set
    End Property

    Public Property PackageSize As String
        Get
            Return _PackageSize
        End Get
        Set
            _PackageSize = Value
            NotifyPropertyChanged("PackageSize")
        End Set
    End Property

    Public Property PackageDownloadState As String
        Get
            Return _PackageDownloadState
        End Get
        Set
            _PackageDownloadState = Value
            NotifyPropertyChanged("PackageDownloadState")
        End Set
    End Property

    Public Property PackageDownloadDestination As String
        Get
            Return _PackageDownloadDestination
        End Get
        Set
            _PackageDownloadDestination = Value
            NotifyPropertyChanged("PackageDownloadDestination")
        End Set
    End Property

End Class

File: PSMultiTools/Classes/PKGDownloadListViewItem.cs
using System.Collections.Generic;
using System.ComponentModel;
using System.Net;
using System.Runtime.CompilerServices;

namespace PSMultiTools.Classes
{

    public class PKGDownloadListViewItem : INotifyPropertyChanged
    {

        private string? _PackageTitleID;
        private string? _PackageName;
        private string? _PackageSize;
        private string? _PackageDownloadState;
        private WebClient? _AssociatedWebClient;
        private string? _PackageContentID;
        private string? _PackageDownloadDestination;

        public WebClient? AssociatedWebClient
        {
            get => _AssociatedWebClient;
            set => SetProperty(ref _AssociatedWebClient, value);
        }

        public string? PackageContentID
        {
            get => _PackageContentID;
            set => SetProperty(ref _PackageContentID, value);
        }

        public string? PackageTitleID
        {
            get => _PackageTitleID;
            set => SetProperty(ref _PackageTitleID, value);
        }

        public string? PackageName
        {
            get => _PackageName;
            set => SetProperty(ref _PackageName, value);
        }

        public string? PackageSize
        {
            get => _PackageSize;
            set => SetProperty(ref _PackageSize, value);
        }

        public string? PackageDownloadState
        {
            get => _PackageDownloadState;
            set => SetProperty(ref _PackageDownloadState, value);
        }

        public string? PackageDownloadDestination
        {
            get => _PackageDownloadDestination;
            set => SetProperty(ref _PackageDownloadDestination, value);
        }

        public event PropertyChangedEventHandler? PropertyChanged;

        protected bool SetProperty<T>(ref T field, T value, [CallerMemberName] string? propertyName = null)
        {
            if (EqualityComparer<T>.Default.Equals(field, value))
                return false;

            field = value;
            OnPropertyChanged(propertyName);
            return true;
        }

        protected void OnPropertyChanged([CallerMemberName] string? propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

    }
}